df.d_init.xml

```diff
@@ -64,95 +136,46 @@
-            <int32_t name='number_of_lower_elevations_shown' since='v0.50.01'/>
-        </compound>
-
-        <compound name='adventure'> bay12: d_init_adventurest
-            <df-flagarray name='flags' index-enum='d_init_flags2'/>
-            <int32_t name='display_length'/>
-            <enum name='z_view_display_type' type-name='d_init_adventurer_zview'/>
-            <int16_t name='z_view_size'/>
-        </compound>
-
-        <compound name='dwarf'> bay12: d_init_dwarfst
-            <int32_t name="population_cap"/>
-            <int32_t name="strict_population_cap"/>
-            <int32_t name="baby_cap_absolute"/>
-            <int32_t name="baby_cap_percent"/>
-            <int32_t name="visitor_cap"/>
-
-            <int32_t name="specific_seed_cap"/>
-            <int32_t name="fortress_seed_cap"/>
-
-            <static-array name='path_cost' type-name='int32_t' count='4'/>
-            <static-array name='embark_rect' type-name='int32_t' count='2'/>
-
-            <compound name='store_dist'>
-                <int32_t name='item_decrease'/>
-                <int32_t name='seed_combine'/>
-                <int32_t name='bucket_combine'/>
-                <int32_t name='barrel_combine'/>
-                <int32_t name='bin_combine'/>
-            </compound>
-
-            <int32_t name='graze_coefficient' since='v0.40.13'/>
-            <int32_t name='maximum_embark_dim' since='v0.50.01'/>
-            <int32_t name='cull_dead_units_at' since='v0.50.06'/>
-        </compound>
+    <struct-type type-name='d_init_featurest'>
+        <df-flagarray name='flags' index-enum='d_init_flags4'/>
+        <enum name="post_prepare_embark_confirmation" type-name='d_init_embark_confirm'/>
+        <enum name="autosave" type-name='d_init_autosave'/>
+    </struct-type>
 
-        <compound name='feature'> bay12: d_init_featurest
-            <df-flagarray name='flags' index-enum='d_init_flags4'/>
-            <enum name="post_prepare_embark_confirmation" type-name='d_init_embark_confirm'/>
-            <enum name="autosave" type-name='d_init_autosave'/>
-        </compound>
+    <bitfield-type type-name='announcement_flags' base-type='uint32_t'> bay12: INIT_ANNOUNCEMENT_FLAG_*
+        <flag-bit name='DO_MEGA' comment='BOX'/>
+        <flag-bit name='PAUSE' comment='P'/>
+        <flag-bit name='RECENTER' comment='R'/>
+        <flag-bit name='A_DISPLAY' comment='A_D'/>
+        <flag-bit name='D_DISPLAY' comment='D_D'/>
+        <flag-bit name='UNIT_COMBAT_REPORT' comment='UCR'/>
+        <flag-bit name='UNIT_COMBAT_REPORT_ALL_ACTIVE' comment='UCR_A'/>
+        <flag-bit name='ALERT' comment='ALERT' since='v0.50.01'/>
+    </bitfield-type>
+
+    <struct-type type-name='announcements' original-name='d_init_announcementst'>
+        <static-array name='flags' type-name='announcement_flags'
+                      count='354' index-enum='announcement_type'/>
```

Comment:
should this be `count='356'` to match the enum range of `announcement_type`?

Reply:
Yes, it should - it looks like that was incorrect in the previous structures as well.